source/beamline/BioXAS/BioXASBeamline.h

```diff
@@ -165,10 +162,16 @@ class BioXASBeamline : public CLSBeamline
 
 	/// Returns the scaler.
 	virtual CLSSIS3820Scaler* scaler() const { return 0; }
+	/// Returns the I0 Keithley amplifier.
+	virtual CLSKeithley428* i0Keithley() const { return 0; }
```

Comment:
I have a question here. Does the subclass must re-implement these functions? 
If yes, it will be better to define those functions as pure virtual functions. In that way, the compiler will tell us if we forget to implement those functions in the subclass.